core/src/main/java/nl/nn/adapterframework/http/rest/ApiServiceDispatcher.java

```diff
@@ -180,7 +180,12 @@ protected JsonObject generateOpenApiJsonSchema(Collection<ApiDispatchConfig> cli
 						methodBuilder.add("summary", adapter.getDescription());
 					}
 					methodBuilder.add("operationId", adapter.getName());
-
+					PipeLine pipeline = adapter.getPipeLine();
+					Json2XmlValidator validator = getJsonValidator(pipeline);
+					if(validator != null && StringUtils.isNotEmpty(validator.getRoot())) {
+						MediaTypes consumes = MediaTypes.valueOf(listener.getConsumes());
+						methodBuilder.add("requestBody", mapRequest(consumes, validator.getRoot()));
+					}
 					//ContentType may have more parameters such as charset and formdata-boundry
 					MediaTypes produces = MediaTypes.valueOf(listener.getProduces());
```

Comment:
In the spirit of 'we want enums', could you refactor the ApiListener a bit?
Replace the `getProduces()` and `getConsumes` with `getProducesEnum()` and `getConsumesEnum`. So you don't have to convert the strings back to enums on line 186 and 190?